force-app/main/default/lwc/serviceDeliveryRow/serviceDeliveryRow.js

```diff
@@ -324,25 +398,76 @@ export default class ServiceDeliveryRow extends LightningElement {
         modal.hide();
     }
 
-    enableFields() {
+    enableDisableFields() {
         this.localFieldSet.forEach(element => {
-            if (element.apiName !== this.fields.contact.fieldApiName) {
+            if (this.hasContact && element.apiName !== this.fields.contact.fieldApiName) {
+                element.disabled = false;
+            } else if (
+                this.hasContact &
+                (element.apiName === this.fields.contact.fieldApiName)
+            ) {
+                element.disabled = true;
+            }
+
+            if (
+                !this.hasContact &&
+                this.hasProgramEngagement &&
+                element.apiName !== this.fields.programEngagement.fieldApiName
+            ) {
+                element.disabled = false;
+            } else if (
+                !this.hasContact &&
+                this.hasProgramEngagement &&
+                element.apiName === this.fields.programEngagement.fieldApiName
+            ) {
+                element.disabled = true;
+            }
+
+            if (
+                !this.hasContact &&
+                !this.hasProgramEngagement &&
+                element.apiName !== this.fields.service.fieldApiName
+            ) {
                 element.disabled = false;
+            } else if (
+                !this.hasContact &&
+                !this.hasProgramEngagement &&
+                element.apiName === this.fields.service.fieldApiName
+            ) {
+                element.disabled = true;
             }
         });
     }
 
     enableServiceInputWithOptions() {
         let result = [];
-        this._filteredValues[SERVICES].forEach(element => {
-            if (element.program === this._targetProgram) {
-                result.push(element);
-            }
-        });
+
+        if (this._filteredValues) {
+            this._filteredValues[SERVICES].forEach(element => {
+                if (element.program === this._targetProgram) {
+                    result.push(element);
+                }
+            });
+        }
+
+        if (this._filteredServices) {
+            result = this._filteredServices;
+        }
+
+        if (this.isServiceFiltered) {
+            this.localFieldSet = JSON.parse(JSON.stringify(this.localFieldSet));
```

Comment:
Looks like we're cloning here as well.  Again, wonder if there is a single place we can do things like this.